admin/app.js

```diff
@@ -210,19 +212,34 @@ passport.deserializeUser(function (user, done) {
   done(null, user)
 })
 
-// passport with custom strategy
-passport.use(new CustomStrategy(
-  require('./authentication/nca-tools-authentication-strategy')
-))
-
-// Passport with local strategy
-passport.use(
-  new LocalStrategy({
-    passReqToCallback: true
-  },
-  require('./authentication/local-strategy')
-  )
-)
+if (config.Auth.mode === authModes.dfeSignIn) {
+  dfeSignInStrategy.initialiseAsync()
+    .then((strategy) => {
+      passport.use(authModes.dfeSignIn, strategy)
+    })
+    .catch((error) => {
+      logger.error(`unable to configure passport for dfeSignin:${error.message}`)
+      process.exit(1)
```

Comment:
Let's test this under load later on and see how it works out.  My concern is that at high-loads we could see our dependency (the Sign-in service) causing our app to fail.

I would prefer our app to handle lack of availability on the Sign-in service by doing something reasonable like showing the user a message asking them to try again later.  Getting that into the start-up phase seems like a lot more work if it's even possible at all.

This is fine for a version 1 though.